### PR TITLE
update travis with matrix and switch serverspec to inspec

### PR DESCRIPTION
**Pull request by aboe76**
Fix the travis button,
Update travis with build matrix
Update kitchen.yml to new version
Switch from serverspec to inspec.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,9 +1,22 @@
...
 install:
-  - export BUNDLE_GEMFILE=$PWD/Gemfile
   - bundle install
 
-script: bundle exec kitchen verify ${INSTANCE}
+env:
+  matrix:
+    - INSTANCE: centos-7
+    - INSTANCE: debian-9
+    - INSTANCE: ubuntu-1804
```
Comment (myii) [inline .travis.yml]: How about introducing the larger matrix we're working towards, i.e. https://github.com/saltstack-formulas/template-formula/pull/38?

Perhaps best to leave this for another a PR...

File: .travis.yml [hunk 2]
```diff
@@ -1,9 +1,22 @@
...
+
+script: 
+  - bundle exec kitchen verify ${INSTANCE}
+
+before_deploy:
+  # Only deploy if the build passed successfully
+  - test $TRAVIS_TEST_RESULT = 0
```
Comment (myii) [inline .travis.yml]: This section has already been removed from the `template-formula`, see https://github.com/saltstack-formulas/template-formula/pull/41/files#diff-354f30a63fb0907d4ad57269548329e3L20.  It has been replaced by defining the `stages` at the top of the linked file.

File: kitchen.yml [hunk 3]
```diff
@@ -6,35 +8,42 @@ driver_config:
...
+        - echo "LANG=en_US.UTF-8" >> /etc/locale.conf
+        - localedef -i en_US -c -f UTF-8 -A /usr/share/locale/locale.alias en_US.UTF-8
       platform: rhel
       run_command: /usr/lib/systemd/systemd
       pid_one_command: /usr/lib/systemd/systemd
   - name: debian-9
     driver_config:
       provision_command:
-        - apt-get update && apt-get install -y locales
+        - apt-get update  && apt-get install -y locales
```
Comment (myii) [inline kitchen.yml]: Extra space introduced by mistake.

File: kitchen.yml [hunk 4]
```diff
@@ -6,35 +8,42 @@ driver_config:
...
         - locale-gen en_US.UTF-8
+      platform: debian
       run_command: /lib/systemd/systemd
       pid_one_command: /lib/systemd/systemd
-  - name: ubuntu-17.10
+  - name: ubuntu-18.04
     driver_config:
       provision_command:
-        - apt-get update && apt-get install -y locales
+        - apt-get update  && apt-get install -y locales && rm -rf /var/lib/apt/lists/*
```
Comment (myii) [inline kitchen.yml]: Extra space introduced by mistake.

**Comment by myii** [comment]
@aboe76 That's a quick eyeball review.  Looks great in general.  When I get a chance, I'll take it for a spin and give you better feedback.  At least there some food for thought in the meanwhile.

**Comment by aboe76** [comment]
only debian 8 fails on libnss-resolve because it's in backports...

**Comment by myii** [comment]
@aboe76 Taking shape nicely!  I've kicked off the `opensuse-leap` job again -- failed at an unexpected place last run.  I understand the simplification in 16c5ec4 but that seems a shame since we want to have a testing suite that runs specifically for each OS.  I'd like to hope we get to the stage where every PR includes specific tests **as a condition of acceptance**.

And as I come around to finishing this comment, the `opensuse-leap` test has also passed.  So merging this.

**Comment by aboe76** [comment]
@myii thanks for this.

**Comment by aboe76** [comment]
next stage update tofs

**Comment by myii** [comment]
@aboe76 Let's finish that in `template-formula` first, so that it can be based on the new changes.

**Comment by myii** [comment]
@aboe76 Keeps failing unexpectedly with:

```
/home/travis/.rvm/gems/ruby-2.4.1/gems/net-scp-1.2.1/lib/net/scp.rb:365:in `block (3 levels) in start_command': SCP did not finish successfully (255):  (Net::SCP::Error)
```

* Does work (eventually) after restarting enough times.

**Pull request merged**